### ISSUE TITLE
Rounding errors with Color.hsl().string()

### ISSUE THREAD

**Issue by s2tephen** [Qix-/color#127]
Certain colors (I've only seen this in HSL so far) don't round correctly when calling `string()`.

For example:

```
var Color = require("color");

var pink = Color('#FF08C2');
console.log(pink.hsl().string()); // 'hsl(314.79999999999995, 100%, 51.6%)'
```

**Comment by jacksenior** [comment]
Observed when using RGBA
`color('#000').fade(0.9).string();`
returns `"rgba(0, 0, 0, 0.09999999999999998)"`

**Comment by Qix-** [comment]
Which version of `color`?

**Comment by henopied** [comment]
@Qix- This issue can be observed in node 9.8.0 and with color 3.0.0.

**Comment by Qix-** [comment]
@henopied Aware it's a rounding problem. I thought it was fixed in `color-string`, though. Let me investigate.

**Comment by astrotim** [comment]
Any news on this issue?

**Comment by iamgoodbytes** [comment]
Possibly related. Alpha value don't seem to get rounded 

`rgba(47, 64, 85, 0.0784313725490196)`
`hsla(213, 29%, 26%, 0.0784313725490196)`

I agree that my input values might be a bit weird but the figma API is returning them unrounded as well.

**Comment by Slapbox** [comment]
This feature would be really great. I don't know why fading a color by `0.8` returns a color with a `valpha` of `0.19999999999999996`, but when you've got a lot of colors that end up like this it gets pretty messy to look at.

`fade(0.8) >>> valpha = 0.19999999999999996`
`fade(0.7) >>> valpha = 0.30000000000000004`

Everything else I've tried has returned a round number as expected.

**Comment by Qix-** [comment]
@Slapbox please read [this article](https://modernweb.com/what-every-javascript-developer-should-know-about-floating-points/) before complaining about your particular case of rounding. That's not what the OP is facing, per se.

**Comment by Slapbox** [comment]
@Qix it's a rounding error nonetheless. Would you prefer I open a new issue then?

**Comment by Qix-** [comment]
@Slapbox No, because the rounding error you're reporting is only fixable by the IEEE. Feel free to open a bug report regarding the IEEE-754 standard.

**Comment by Slapbox** [comment]
@Qix- I understand calculations are done with floating points, but surely you're not telling me that we can't round `0.19999999999999996` when we're done with calculations and getting a string value with `.rgb().string()`.

**Comment by Qix-** [comment]
@slapbox I'm not sure why you're pushing so hard on this. Yes it can be rounded, I just want to make sure you understand where your errors are coming from as opposed to OP's.

**Comment by Slapbox** [comment]
I'm not, "pushing hard." I reported an issue and got what I perceived to be and condescending response calling my report "complaining" and then another suggesting that rounding the output is impossible and I need to educate myself.

**Comment by gordonnl** [comment]
Have their been any decisions made on this issue? Still occurring in 3.1.3

**Comment by bbugh** [comment]
It's not just alpha, `hue` is not rounded in certain conditions. For example, converting `tomato` to hsl even with `round()` or `round(1)` or no round results in unrounded hue:

```javascript
Color('tomato').hsl().round().string()  === "hsl(9.100000000000023, 100%, 63.9%)"
Color('tomato').hsl().round(1).string() === "hsl(9.100000000000023, 100%, 63.9%)"
Color('tomato').hsl().round(5).string() === "hsl(9.100000000000023, 100%, 63.9%)"
Color('tomato').hsl().string()          === "hsl(9.100000000000023, 100%, 63.9%)"
```

**Comment by stevage** [comment]
The bug is that the library is taking a floating point number, using `toFixed()` to convert it to a rounded number in string form, then immediately converting back into a floating point number again.

```js
		return new Color(this.color.map(roundToPlace(places)).concat(this.valpha), this.model);
```

and 

```js
let self = this.model in colorString.to ? this : this.rgb();
self = self.round(typeof places === 'number' ? places : 1);
const args = self.valpha === 1 ? self.color : self.color.concat(this.valpha);
return colorString.to[self.model](args);
```

The rounding needs to happen at the last minute when the result is passed back to the caller of `.string()` or `.hex()` etc.

(Nothing to do with the IEEE.)

**Comment by Sidnioulz** [comment]
@Qix- PR open! https://github.com/Qix-/color/pull/271

I'd like to either export `roundColor` or move it to its own file so it can be exported to a test file but hidden from the NPM package public API. That would allow me to add unit tests covering all possible cases of trailing zeros (e.g. 1.0, 1.20, 1.02, 1.020, etc.). But I feel this would significantly change how you bundle and distribute this package. Any preference on how to proceed?

**Comment by Sidnioulz** [comment]
Hi @Qix-, taking the liberty to ping you to see if you could please find time for another review. Thanks!
